xml/src/main/java/org/ehcache/config/xml/XmlConfiguration.java

```diff
@@ -146,6 +146,9 @@ public XmlConfiguration(URL url, final ClassLoader classLoader)
    */
   public XmlConfiguration(URL url, final ClassLoader classLoader, final Map<String, ClassLoader> cacheClassLoaders)
       throws ClassNotFoundException, SAXException, InstantiationException, IllegalAccessException, IOException {
+    if(url == null) {
+      throw new NullPointerException("The resource specified by the URL does not exist");
```

Comment:
That's not the correct message. The message should be: "Url cannot be null".
And while you are at it, I believe null in the following two parameters is an error also.